### PR TITLE
mds: Fix 8MMDSPing is not an MMDSOp type

### PR DESCRIPTION
**Pull request by joscollin**

### CODE REVIEW

**Comment by batrick** [comment]
Marking this DNM because the fix is incomplete.

**Comment by joscollin** [comment]
http://pulpito.front.sepia.ceph.com/jcollin-2020-07-13_06:13:24-fs-wip-mmdsping-fix-distro-basic-smithi/

**Comment by batrick** [comment]
jenkins retest this please

**Comment by batrick** [comment]
jenkins test dashboard backend

**Comment by batrick** [comment]
@joscollin status?

**Comment by joscollin** [comment]
@batrick There are [no related failures](https://github.com/ceph/ceph/pull/36004#issuecomment-657970434). So please merge this.

Please note: This PR only prevents unnecessary test failures, so that the current tests could continue. The issue mentioned in the tracker (why the test continues after the assert) is yet to be investigated.

**Comment by batrick** [comment]
> @batrick There are [no related failures](https://github.com/ceph/ceph/pull/36004#issuecomment-657970434). So please merge this.

The fix is still incomplete. As we discussed, you need to understand why the `MMDSPing` message was processed by `MDSRank::ms_dispatch` and not in earlier testing. Check the logs.

> Please note: This PR only prevents unnecessary test failures, so that the current tests could continue. The issue mentioned in the tracker (why the test continues after the assert) is yet to be investigated.

**Comment by joscollin** [comment]
Closing this. Will send a new PR to fix the issue mentioned in the tracker.

**Comment by batrick** [comment]
@joscollin I've been including this in my integration branch so I don't see the failure. So I'm reopening it.

**Comment by joscollin** [comment]
Added one more inter-mds class from https://github.com/ceph/ceph/pull/26004.

**Comment by batrick** [comment]
@joscollin here's an instance of the failure:

/ceph/teuthology-archive/pdonnell-2020-09-18_04:24:45-fs-wip-pdonnell-testing-20200918.022351-distro-basic-smithi/5445285/remote/smithi140/log/ceph-mds.a.log.gz

Please save the log and debug.

**Comment by joscollin** [comment]
@batrick Thanks Patrick! I'm checking that now.

File: src/mds/MDSDaemon.cc [hunk 1]
```diff
@@ -462,25 +462,25 @@ void MDSDaemon::clean_up_admin_socket()
...
+  dout(10) << sizeof(elist<void*>::item) << "\t elist<>::item * 2 = " << 2*sizeof(elist<void*>::item) << dendl;
   dout(10) << sizeof(fnode_t) << "\t fnode_t " << dendl;
-  dout(10) << sizeof(nest_info_t) << "\t  nest_info_t *2" << dendl;
-  dout(10) << sizeof(frag_info_t) << "\t  frag_info_t *2" << dendl;
-  dout(10) << sizeof(Capability) << "\tCapability " << dendl;
-  dout(10) << sizeof(xlist<void*>::item) << "\t xlist<>::item   *2=" << 2*sizeof(xlist<void*>::item) << dendl;
+  dout(10) << sizeof(nest_info_t) << "\t nest_info_t" << dendl;
+  dout(10) << sizeof(frag_info_t) << "\t frag_info_t" << dendl;
+  dout(10) << sizeof(Capability) << "\t Capability " << dendl;
+  dout(10) << sizeof(xlist<void*>::item) << "\t xlist<>::item * 2 = " << 2*sizeof(xlist<void*>::item) << dendl;
```
Comment (batrick) [inline src/mds/MDSDaemon.cc:483]: Please move this commit to another PR.

**Review by batrick — CHANGES_REQUESTED** [review]

**Comment by joscollin** [comment]
http://pulpito.front.sepia.ceph.com/jcollin-2020-10-20_06:44:34-fs-wip-mmdsping-fix-distro-basic-smithi/

`fs` suite ran with `--limit 100`

**Review by batrick — APPROVED** [review]

**Review by batrick — CHANGES_REQUESTED** [review]
Actually, there's one more change to make.

`MDSRank::is_valid_message` needs to recognize these `MMDSPing` and `MMDSMetrics` messages and not return true. We don't want the MDSRank dispatcher to ever process them.

**Comment by joscollin** [comment]
> Actually, there's one more change to make.
> 
> `MDSRank::is_valid_message` needs to recognize these `MMDSPing` and `MMDSMetrics` messages and not return true. We don't want the MDSRank dispatcher to ever process them.

Right, I'll check that code path and make the change.

**Comment by joscollin** [comment]
> Actually, there's one more change to make.
> 
> `MDSRank::is_valid_message` needs to recognize these `MMDSPing` and `MMDSMetrics` messages and not return true. We don't want the MDSRank dispatcher to ever process them.

I've checked the code once again today morning and I think this is unnecessary, because:

1. `MDSRank::is_valid_message` currently returns `false` for an incoming `MSG_MDS_PING` and `MSG_MDS_METRICS`  and thus `MDSRank::_dispatch` doesn't process the message.
2. In such cases, an incoming `MSG_MDS_PING` or an `MSG_MDS_METRICS` would assert [here](https://github.com/ceph/ceph/blob/master/src/mds/MDSRank.cc#L1016) at the first place as it is mds-mds message?

**Comment by joscollin** [comment]
Sorry, I've edited my comment ^

**Comment by joscollin** [comment]
jenkins test make check

**Comment by batrick** [comment]
Right, my mistake. Okay PR looks good as-is.

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201021.232150

failures unrelated

**Pull request merged**